steampipecloud/table_steampipecloud_workspace_snapshot.go

```diff
@@ -0,0 +1,597 @@
+				Transform:   transform.FromQual("query_where"),
+			},
+			{
+				Name:        "created_at",
+				Description: "The time when the snapshot was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "created_by_id",
+				Description: "ID of the user who created the snapshot.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "created_by",
+				Description: "Information about the user who created the snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "updated_at",
+				Description: "The time when the snapshot was last updated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "updated_by_id",
+				Description: "ID of the user who last updated the snapshot.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "updated_by",
+				Description: "Information about the user who last updated the snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "version_id",
+				Description: "The current version ID for the snapshot.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromCamel(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listWorkspaceSnapshots(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var workspace *openapi.Workspace
+	switch w := h.Item.(type) {
+	case openapi.Workspace:
+		wo := h.Item.(openapi.Workspace)
+		workspace = &wo
+	case *openapi.Workspace:
+		workspace = h.Item.(*openapi.Workspace)
+	default:
+		plugin.Logger(ctx).Error("listWorkspaceSnapshots", "unknown response type for workspace list parent hydrate call", w)
+	}
+
+	// If the requested number of items is less than the paging max limit
+	// set the limit to that instead
+	maxResults := int32(100)
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < int64(maxResults) {
+			if *limit < 1 {
+				maxResults = int32(1)
+			} else {
+				maxResults = int32(*limit)
+			}
+		}
+	}
+
+	var err error
+	if strings.HasPrefix(workspace.IdentityId, "u_") {
+		err = listUserWorkspaceSnapshots(ctx, d, h, workspace.IdentityId, workspace.Handle, maxResults)
+	} else {
+		err = listOrgWorkspaceSnapshots(ctx, d, h, workspace.IdentityId, workspace.Handle, maxResults)
+	}
+
+	if err != nil {
+		plugin.Logger(ctx).Error("listWorkspaceSnapshots", "error", err)
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func listUserWorkspaceSnapshots(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, userHandle string, workspaceHandle string, maxResults int32) error {
+	// Create Session
+	svc, err := connect(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("listUserWorkspaceSnapshots", "connection_error", err)
+		return err
+	}
+
+	var filter string
+	// If query_where is passed in the query, that takes precedence
```

Comment:
@siddharthaturbot In other tables we take this approach, but moving forward we're changing so `query_where` should be combined with the other quals instead.

Are there other tables in this plugin where we have an explicit filter column, and if so, do you know how those tables behave?

Reply:
Ok thanks @siddharthaturbot , can you please make the change then to make the query concat with other quals' filters?